assets/js/base/components/price-slider/index.js

```diff
@@ -0,0 +1,405 @@
+/**
+ * External dependencies
+ */
+import { sprintf, __ } from '@wordpress/i18n';
+import { Fragment, useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { useDebounce } from '@woocommerce/base-hooks';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { constrainRangeSliderValues, formatCurrencyForInput } from './utils';
+
+const PriceSlider = ( {
+	initialMin,
+	initialMax,
+	min,
+	max,
+	onChange,
+	step,
+	currencySymbol,
+	priceFormat,
+	showInputFields,
+	showFilterButton,
+	isLoading,
+} ) => {
+	const [ value, setValue ] = useState( {
+		min: initialMin,
+		max: initialMax,
+	} );
+	const [ formattedValue, setFormattedValue ] = useState( {
+		min: formatCurrencyForInput( value.min, priceFormat, currencySymbol ),
+		max: formatCurrencyForInput( value.max, priceFormat, currencySymbol ),
+	} );
```

Comment:
This might work better implementing `useMemo` here as it could automatically recalculate the formatted values when the state (or priceFormat, or currencySymbol) changes.

```js
const [ formattedMinPrice, formattedMaxprice ] = useMemo( () => {
   return [
        formatCurrencyForInput( minPrice, priceFormat, currencySymbol ),
        formatCurrencyForInput( maxPrice, priceFormat, currencySymbol )
   ];
}, [ minPrice, maxPrice, priceFormat, currencySymbol ] );
```

It will simplify the code a bit more and remove the necessity to update formatted price values everywhere.